### ISSUE TITLE
panic when building with go 1.22 and wails 2.7.1 on Windows

### ISSUE THREAD

**Issue by acastruc** [wailsapp/wails#3242]
### Description

On a fresh Windows VM, I installed [Go 1.22 ](https://tip.golang.org/doc/go1.22) , then attempted to build a project that used wails v2.7.1. I ran `wails build [-clean]` and got a panic message.

```
> wails build
Wails CLI v2.7.1
# Build Options

Platform(s)        | windows/amd64
Compiler           | C:\Program Files\Go\bin\go.exe
Skip Bindings      | false
Build Mode         | production
Devtools           | false
Frontend Directory | C:\Users\me\app\frontend
Obfuscated         | false
Skip Frontend      | false
Compress           | false
Package            | true
Clean Bin Dir      | false
LDFlags            |
Tags               | []
Race Detector      | false

# Building target: windows/amd64

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1259f9e]

goroutine 15 [running]:
go/types.(*Checker).handleBailout(0xc0000dbc00, 0xc00045bbd0)
        C:/Program Files/Go/src/go/types/check.go:367 +0x88
panic({0x1b259c0?, 0x2869b70?})
        C:/Program Files/Go/src/runtime/panic.go:770 +0x132
go/types.(*StdSizes).Sizeof(0x0, {0x1fb8db8, 0x28772e0})
        C:/Program Files/Go/src/go/types/sizes.go:228 +0x31e
go/types.(*Config).sizeof(...)
        C:/Program Files/Go/src/go/types/sizes.go:333
go/types.representableConst.func1({0x1fb8db8?, 0x28772e0?})
        C:/Program Files/Go/src/go/types/const.go:76 +0x9e
go/types.representableConst({0x1fbbfa0, 0x262a9e8}, 0xc0000dbc00, 0x28772e0, 0xc0004595d0)
        C:/Program Files/Go/src/go/types/const.go:92 +0x192
go/types.(*Checker).representation(0xc0000dbc00, 0xc000463280, 0x28772e0)
        C:/Program Files/Go/src/go/types/const.go:256 +0x65
go/types.(*Checker).implicitTypeAndValue(0xc0000dbc00, 0xc000463280, {0x1fb8db8, 0x28772e0})
        C:/Program Files/Go/src/go/types/expr.go:375 +0x30d
go/types.(*Checker).assignment(0xc0000dbc00, 0xc000463280, {0x1fb8db8, 0x28772e0}, {0xc000332bd0, 0x14})
        C:/Program Files/Go/src/go/types/assignments.go:52 +0x2e5
go/types.(*Checker).arguments(0xc0000dbc00, 0xc001367b00, 0xc000463240, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, {0xc001175920, ...}, ...)
        C:/Program Files/Go/src/go/types/call.go:654 +0x13dc
go/types.(*Checker).callExpr(0xc0000dbc00, 0xc000463200, 0xc001367b00)
        C:/Program Files/Go/src/go/types/call.go:304 +0x6e9
go/types.(*Checker).exprInternal(0xc0000dbc00, 0x0, 0xc000463200, {0x1fbadd0, 0xc001367b00}, {0x0, 0x0})
        C:/Program Files/Go/src/go/types/expr.go:1374 +0xf8
go/types.(*Checker).rawExpr(0xc0000dbc00, 0x0, 0xc000463200, {0x1fbadd0?, 0xc001367b00?}, {0x0?, 0x0?}, 0x0)
        C:/Program Files/Go/src/go/types/expr.go:979 +0x19e
go/types.(*Checker).stmt(0xc0000dbc00, 0x0, {0x1fba980, 0xc00132f0b0})
        C:/Program Files/Go/src/go/types/stmt.go:394 +0x12d
go/types.(*Checker).stmtList(0xc0000dbc00, 0x0, {0xc0003ef300?, 0x0?, 0x2e?})
        C:/Program Files/Go/src/go/types/stmt.go:121 +0x85
go/types.(*Checker).funcBody(0xc0000dbc00, 0x1fb8e08?, {0xc000236d10?, 0xc00017cc40?}, 0xc00042b7c0, 0xc000e89aa0, {0x0?, 0x0?})
        C:/Program Files/Go/src/go/types/stmt.go:41 +0x331
go/types.(*Checker).funcDecl.func1()
        C:/Program Files/Go/src/go/types/decl.go:852 +0x3a
go/types.(*Checker).processDelayed(0xc0000dbc00, 0x0)
        C:/Program Files/Go/src/go/types/check.go:467 +0x162
go/types.(*Checker).checkFiles(0xc0000dbc00, {0xc0002f33e0, 0x4, 0x4})
        C:/Program Files/Go/src/go/types/check.go:411 +0x1cc
go/types.(*Checker).Files(...)
        C:/Program Files/Go/src/go/types/check.go:372
golang.org/x/tools/go/packages.(*loader).loadPackage(0xc00014c380, 0xc0004164b0)
        C:/Users/me/go/pkg/mod/golang.org/x/tools@v0.6.0/go/packages/packages.go:1052 +0xa72
golang.org/x/tools/go/packages.(*loader).loadRecursive.func1()
        C:/Users/me/go/pkg/mod/golang.org/x/tools@v0.6.0/go/packages/packages.go:851 +0x1a9
sync.(*Once).doSlow(0xc00055df98?, 0x0?)
        C:/Program Files/Go/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        C:/Program Files/Go/src/sync/once.go:65
golang.org/x/tools/go/packages.(*loader).loadRecursive(0x1?, 0xc00055dfd0?)
        C:/Users/me/go/pkg/mod/golang.org/x/tools@v0.6.0/go/packages/packages.go:839 +0x4a
golang.org/x/tools/go/packages.(*loader).refine.func2(0x0?)
        C:/Users/me/go/pkg/mod/golang.org/x/tools@v0.6.0/go/packages/packages.go:774 +0x26
created by golang.org/x/tools/go/packages.(*loader).refine in goroutine 1
        C:/Users/me/go/pkg/mod/golang.org/x/tools@v0.6.0/go/packages/packages.go:773 +0xccf
```

### To Reproduce

1. Install Go 1.22 https://tip.golang.org/doc/go1.22
2. Install wails v2.7.1 and all necessary dependencies
3.  I used an existing wails project (cannot share source) I am working on but I bet it would work on a new vanilla project from template
4. Run `wails build [-clean]` from project root.
result: panic
I then installed go 1.21.x, problem still occurred. I verified go version with `go version`
I searched my HD and only found the 1 go binary (1.21)
I found it curious that wails doctor still indicated it was Go 1.22.
As a last resort I had to delete wails and reinstall before it worked again.

### Expected behaviour

The wails build command should generate a binary as normal with go1.22. No panic.

### Screenshots

<img width="680" alt="image" src="https://github.com/wailsapp/wails/assets/5711953/9b1f03c2-33ec-4f2a-afff-d3a07e59dd7e">

### Attempted Fixes

I installed go 1.21, ran build command again. Panic still occurred.
Verified go version with `go version` -> 1.21 
I ran wails doctor and it indicated go 1.22 ? This puzzled me.
Had to reinstall wails before build worked. 

### System Details

```shell
wails doctor


          Wails Doctor          
                                


# Wails
Version | v2.7.1

# System
┌───────────────────────────────────────────────────────────────────────────────────────┐
| OS           | Windows 10 Enterprise                                                  |
| Version      | 2009 (Build: 22000)                                                    |
| ID           | 21H2                                                                   |
| Go Version   | go1.22.0                                                               |
| Platform     | windows                                                                |
| Architecture | amd64                                                                  |
| CPU          | Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz                          |
| GPU 1        | Microsoft Hyper-V Video (Microsoft) - Driver: 10.0.22000.1             |
| GPU 2        | Microsoft Remote Display Adapter (Microsoft) - Driver: 10.0.22000.2600 |
| Memory       | 4GB                                                                    |
└───────────────────────────────────────────────────────────────────────────────────────┘

# Dependencies
┌───────────────────────────────────────────────────────┐
| Dependency | Package Name | Status    | Version       |
| WebView2   | N/A          | Installed | 121.0.2277.98 |
| Nodejs     | N/A          | Installed | 20.11.0       |
| npm        | N/A          | Installed | 10.2.4        |
| *upx       | N/A          | Available |               |
| *nsis      | N/A          | Available |               |
└─────────────── * - Optional Dependency ───────────────┘

# Diagnosis
Optional package(s) installation details:
  - upx : Available at https://upx.github.io/
  - nsis : More info at https://wails.io/docs/guides/windows-installer/

 SUCCESS  Your system is ready for Wails development!

 ♥   If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

**Comment by acastruc** [comment]
I could but as I mentioned it was a fresh VM with fresh Go install and fresh wails install.

**Comment by leaanthony** [comment]
I just reproduced it. Absolutely looks like an issue with Go itself. I'll try open an issue on the Go repo today

**Comment by acastruc** [comment]
Ohh (waves hand), I'm no hero...

**Comment by acastruc** [comment]
Been tinkering switching between go versions

> so "go install wails" with 1.21 creates a good wails CLI (build works)
> "go install wails" with 1.22 creates bad wails CLI (panic)
just fyi, not sure if this informs anything you didnt already know.

**Comment by leaanthony** [comment]
Do you use Goland or VSC? Perhaps if we create a new project using the `-ide` flag and load it in the IDE, we could run it in debug mode and hopefully get some context about which Go command is causing this.

**Comment by acastruc** [comment]
@leaanthony both. We have multiple devs on our team, some use Goland on win, some VSCode.

**Comment by stffabi** [comment]
I was also able to reproduce it on macOS. Building `wails` cli with  go1.22.0. It also happens with `wails dev` just after `executing go mod tidy`. Manually building the wails app directly with go works also.

The culprit seems to be `golang.org/x/tools`, after updating it to at least `v0.13.0` the `wails` cli started to be working again, if the cli was built with `go1.22`.

It seems like the issue has already been reported at the go repo https://github.com/golang/go/issues/64812

**Comment by stffabi** [comment]
PR to bump x/tools is up #3244

**Comment by stffabi** [comment]
@acastruc would it be possible for you to give the PR #3244 a try?

**Comment by acastruc** [comment]
@stffabi fix verified, thanks

**Comment by leaanthony** [comment]
Released as part of v2.8 👍
